File: bindings/java/Carbon/Recognition/Speech/SpeechRecognitionResult.java
package Carbon.Recognition.Speech;
//
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//

import Carbon.Recognition.RecognitionStatus;

/// <summary>
/// Defines result of speech recognition.
/// </summary>
public class SpeechRecognitionResult 
{
    protected SpeechRecognitionResult(Carbon.Internal.RecognitionResult result)
    {
        this._ResultId = result.getResultId();
        this._RecognizedText = result.getText();
        this._Reason = RecognitionStatus.values()[result.getReason().swigValue()];
    }

    /// <summary>
    /// Specifies the result identifier.
    /// </summary>
    public String getResultId()
    {
        return _ResultId;
    }// { get; }
    private String _ResultId;

    /// <summary>
    /// Specifies status of the result.
    /// </summary>
    public RecognitionStatus getReason() // { get; }
    {
        return _Reason;
    }
    private RecognitionStatus _Reason;

    /// <summary>
    /// Presents the recognized text in the result.
    /// </summary>
    public String getRecognizedText() {
        return _RecognizedText;
    } // { get; }
    private String _RecognizedText;

    /// <summary>
    /// Returns a String that represents the speech recognition result.
    /// </summary>
    /// <returns>A String that represents the speech recognition result.</returns>
    @Override
    public String toString()
    {
        return "ResultId:" + _ResultId + " Status:" + _Reason + " Recognized text:<" + _RecognizedText + ">.";
    }
}

File: source/bindings/js/src/sdk/SpeechRecognitionResult.ts
//
// copyright (c) Microsoft. All rights reserved.
// See https://aka.ms/csspeech/license201809 for the full license information.
//
import { PropertyCollection, RecognitionResult, ResultReason } from "./Exports";

/**
 * Defines result of speech recognition.
 * @class SpeechRecognitionResult
 */
export class SpeechRecognitionResult extends RecognitionResult {
    /**
     * Creates and initializes an instance of this class.
     * @constructor
     * @public
     * @param {string} resultId - The result id.
     * @param {ResultReason} reason - The reason.
     * @param {string} text - The recognized text.
     * @param {number} duration - The duration.
     * @param {number} offset - The offset into the stream.
     * @param {string} errorDetails - Error details, if provided.
     * @param {string} json - Additional Json, if provided.
     * @param {PropertyCollection} properties - Additional properties, if provided.
     */
    public constructor(resultId?: string, reason?: ResultReason, text?: string, duration?: number, offset?: number, errorDetails?: string, json?: string, properties?: PropertyCollection) {
        super(resultId, reason, text, duration, offset, errorDetails, json, properties);
    }
}
